cpp/src/cugraph.cu

```diff
@@ -113,6 +115,109 @@ gdf_error gdf_adj_list::get_source_indices (gdf_column *src_indices) {
   return GDF_SUCCESS;
 }
 
+gdf_error gdf_renumber_vertices(const gdf_column *src, const gdf_column *dst,
+				gdf_column **src_renumbered, gdf_column **dst_renumbered,
+				gdf_column **numbering_map) {
+
+  GDF_REQUIRE( src->size == dst->size, GDF_COLUMN_SIZE_MISMATCH );
+  GDF_REQUIRE( src->dtype == dst->dtype, GDF_UNSUPPORTED_DTYPE );
+  GDF_REQUIRE( ((src->dtype == GDF_INT32) || (src->dtype == GDF_INT64)), GDF_UNSUPPORTED_DTYPE );
+  GDF_REQUIRE( src->size > 0, GDF_DATASET_EMPTY ); 
+
+  *src_renumbered = new gdf_column;
+  *dst_renumbered = new gdf_column;
+  *numbering_map = new gdf_column;
+
+  //
+  //  TODO: we're currently renumbering without using valid.  We need to
+  //        worry about that at some point, but for now we'll just
+  //        copy the valid pointers to the new columns and go from there.
+  //
+  cudaStream_t stream{nullptr};
+
+  size_t src_size = src->size;
+  size_t new_size;
+
+  //
+  // TODO:  I assume int64_t for output.  A few thoughts:
+  //
+  //    * I could match src->dtype - since if the raw values fit in an int32_t,
+  //      then the renumbered values must fit within an int32_t
+  //    * If new_size < (2^31 - 1) then I could allocate 32-bit integers
+  //      and copy them in order to make the final footprint smaller.
+  //
+  //
+  //  NOTE:  Forcing match right now - it appears that cugraph is artficially
+  //         forcing the type to be 32
+  if (src->dtype == GDF_INT32) {
+    int32_t *tmp;
+
+    printf("in renumber, 32-bit\n");
+    ALLOC_MANAGED_TRY((void**) &tmp, sizeof(int32_t) * src->size, stream);
+    gdf_column_view((*src_renumbered), tmp, src->valid, src->size, src->dtype);
+  
+    ALLOC_MANAGED_TRY((void**) &tmp, sizeof(int32_t) * src->size, stream);
+    gdf_column_view((*dst_renumbered), tmp, dst->valid, dst->size, dst->dtype);
+
+    gdf_error err = cugraph::renumber_vertices(src_size,
+					       (const int32_t *) src->data,
+					       (const int32_t *) dst->data,
+					       (int32_t *) (*src_renumbered)->data,
+					       (int32_t *) (*dst_renumbered)->data,
+					       &new_size, &tmp);
+    if (err != GDF_SUCCESS)
+      return err;
+
+    gdf_column_view((*numbering_map), tmp, nullptr, new_size, src->dtype);
+  } else if (src->dtype == GDF_INT64) {
+
+    //
+    //  NOTE: At the moment, we force the renumbered graph to use
+    //        32-bit integer ids.  Since renumbering is going to make
+    //        the vertex range dense, this limits us to 2 billion
+    //        vertices.
+    //
+    //        The renumbering code supports 64-bit integer generation
+    //        so we can run this with int64_t output if desired...
+    //        but none of the algorithms support that.
+    //
+    printf("in renumber, 64-bit, src->dtype = %d\n", src->dtype);
```

Comment:
A couple of `printf` was added to this file

Reply:
Cleaned up in next push.

Reply:
This should be disabled in the final phase (it is ok to keep them in renumbering for now if you plan to do another PR later)